Utilities/Thread.cpp

```diff
@@ -3028,6 +3030,39 @@ void thread_ctrl::set_native_priority(int priority)
 	{
 		sig_log.error("SetThreadPriority() failed: %s", fmt::win_error{GetLastError(), nullptr});
 	}
+#elif defined(__linux__)
+	// available niceness for nonroot: 0~19
+	// available niceness for root: -20~19
+
+	int linuxprio = 0;
+	id_t threadpid = gettid();
+	uid_t euid = geteuid();
+
+	if (euid == 0)
+	{
+		linuxprio = 0;
+		if (priority > 0)
+			linuxprio = -6;
+		else if (priority < 0)
+			linuxprio = 6;
+	}
+	else
+	{
+		linuxprio = 6;
+		if (priority > 0)
+			linuxprio = 0;
+		else if (priority < 0)
+			linuxprio = 12;
+	}
+
+	// nonroot cannot increase niceness value
+	if ((getpriority(PRIO_PROCESS, threadpid) < linuxprio) || (euid == 0))
```

Comment:
So, if a thread sets lower priority, it cannot increase it afterwards if it is non-root?

Reply:
unfortunately, yep

Reply:
This is a problem because, named_thread is recycled from a pool after use.

Reply:
yes. i noticed this by testing alot and game threads are stuck with low priority. im thinking maybe we only should allow priority changes for root user and disregard nonroots...

Reply:
I think so too.

Reply:
done

Reply:
Does this handle the case where a user might have the appropriate rights (`limits.conf` or `CAP_SYS_NICE`)?